widgetry/src/widgets/button.rs

```diff
@@ -782,8 +783,8 @@ pub struct Image<'a> {
 }
 
 #[derive(Clone, Debug, Default)]
-struct Label<'a> {
-    text: Option<&'a str>,
+struct Label {
+    text: Option<String>,
```

Comment:
Seems like it'd be preferable to keep the old behavior of not requiring a new heap allocation. Does this change enable some improvement elsewhere?

Reply:
I tried, but couldn't get this to work. This is necessary for the API change. If `label_text` takes `AsRef<str>`, there's not a lifetime on it. I tried a few variations, but couldn't figure it out. I'll try again and followup if I can get it.

Reply:
Oh right, I see why that would be hard. I mean, it's nice to avoid requiring heap allocation since a lot of our buttons are static text, but... I assume this is pretty small potatoes compared to other ABStreets perf hot spots.a